output-comparator/src/main/java/org/mobilitydata/gtfsvalidator/outputcomparator/cli/Main.java

```diff
@@ -115,92 +106,45 @@ public static void main(String[] argv) {
         referenceReport = ValidationReport.fromPath(referenceReportPath);
         latestReport = ValidationReport.fromPath(latestReportPath);
       } catch (IOException ioException) {
-        logger.atSevere().withCause(ioException);
+        logger.atSevere().withCause(ioException).log("Error reading validation reports");
         // in case a file is corrupted, add the sourceId to the list of corrupted sources
-        corruptedSources.add(sourceId);
+        corruptedSources.addCorruptedSource(sourceId);
         continue;
       }
-      if (referenceReport.hasSameErrorCodes(latestReport)) {
-        continue;
-      }
-      for (String noticeCode : referenceReport.getNewErrorsListing(latestReport)) {
-        NoticeComparisonReport noticeComparisonReport =
-            acceptanceTestReportMap.getOrDefault(noticeCode, new NoticeComparisonReport());
-        acceptanceTestReportMap.putIfAbsent(noticeCode, noticeComparisonReport);
-        noticeComparisonReport.update(
-            sourceId,
-            latestReport.getErrorNoticeReportByNoticeCode(noticeCode).getTotalNotices(),
-            sourceUrlContainer);
-      }
-      if (referenceReport.getNewErrorsListing(latestReport).size() >= args.getNewErrorThreshold()) {
-        ++badDatasetCount;
-      }
+      newErrors.compareValidationReports(sourceId, sourceUrl, referenceReport, latestReport);
+      droppedErrors.compareValidationReports(sourceId, sourceUrl, latestReport, referenceReport);
```

Comment:
This is great. This also means that we apply the same threshold for both new errors and dropped errors. 

@isabelle-dr Do you see a case where it would be important to have a different threshold for each case? Eg. 10% above for new errors, 20% under for dropped errors. I'm fine with having the same for both cases, just wondering what are your thoughts on this.

Reply:
Interesting question, I believe the current threshold is 1%. 

In the short term: 
I'd like to have a threshold as: every new occurrence of a notice (error or warning) is captured.
I am saying this because I'd like to use this test more for analytics and impact management (and depending on the results: contact every agency concerned), more than a green light to merge a PR or not (this was the initial idea, and this is why it's called "acceptance test").
So I'd like the threshold to not be a % anymore: just a count, and if it is it >1, it appears in the report + message.
This was the idea behind issue #1076.

Longer term if we see a need: a custom threshold that the developer can define depending on the granulometry needed.

If we go this route we might want to:
1.  rename this test (maybe "impact report")
2. run it ad-hoc instead of on every commit, knowing that it would definitely be run before merging as part of the review process.

@bdferris-v2 and @maximearmstrong what do you think of this?

Reply:
And we would use `end_to_end` on every commit just to make sure nothing is broken.

Reply:
I still like the idea of keeping acceptance test with a relatively strict threshold.

The end-to-end tests are really only catching issues where a new change causes the validator to crash, which is not a particularly frequent failure case.  On the other hand, the diff-style acceptance test is capturing any regression in added/removed errors (we should probably add warnings in a follow-up).  I think we should be running this for every change, both for those that are changing notices and those that are not.  It's a good sanity check.

@isabelle-dr to your point about analytics and impact management, what are you hoping to measure here?

Reply:
> I think we should be running this for every change, both for those that are changing notices and those that are not. It's a good sanity check.

Here do you mean for every commit or for every PR? Isabelle and I discussed the idea of requesting the acceptance tests to succeed on the last commit, before a PR is merged, but we are wondering if it is relevant to run them on each commit?

Our plan was to allow the acceptance test to be run "on-demand", by adding `[acceptance test run]` at the end of the commit message for instance, instead of adding `[acceptance test skip]` to skip them. One could run them when they think their code is ready to test. We would then use the end-to-end workflow as a "safety net", but I agree with you that the use case where the validator would crash is uncommon. I've refactored the end-to-end workflow in #1243 so it better fits our needs. Your feedback is welcome.

> I still like the idea of keeping acceptance test with a relatively strict threshold.

I'm fine with this.

Reply:
@bdferris-v2 

> to your point about analytics and impact management, what are you hoping to measure here?

I am hoping to get:
1. Getting general stats on what are the common error and warnings in order to improve the spec and the validator documentation. 
2. Get a list of producers and consumers that are affected by a change in data validity (additional or disappearing errors) and make sure they are aware (for example: by publishing a the list of datasets affected in the release note, and later down the road: set-up a notification system)

I have gathered my thoughts very quickly [here](https://docs.google.com/document/d/1Im8ZVDbGpZu20OzFa9fBY0ubi9y8502xtMxM0aFEsQs/edit#), there are still some questions to explore (how to best let the producers know that a change affects their dataset? How to let consumers know that there is a validity change in a dataset they are consuming?). I hope to include the PR process graph in the contribution guideline so code contributors know what to expect.

I think what we see with the addition of Fares v2 is a good example to put this into practice.

Reply:
I think output comparator, as implemented, will support task #2.  For task #1, I would think you'd just periodically run the full validator on all feeds in the Mobility Database, without needing to compare it against some other version of the validator.

In general, I'm supportive of all of this.  However I do wonder if we are getting beyond the scope of this PR?

Reply:
That makes sense.
You're right: this discussion is getting beyond the scope of this PR 😊, I'll let @maximearmstrong give the final review, thanks for this addition @bdferris-v2!